code/workspaces/client-api/src/schema/resolvers.js

```diff
@@ -103,10 +103,11 @@ const resolvers = {
     id: obj => (obj._id), // eslint-disable-line no-underscore-dangle
     redirectUrl: (obj, args, { token }) => stackUrlService(obj.projectKey, obj, token),
     category: obj => (obj.category ? obj.category.toUpperCase() : null),
-    // TODO - replace this with actual assets owned by the stack
-    assets: () => ([
-      { assetId: 'asset-1', name: 'Test asset 1', version: '0.1', fileLocation: '/file/path1' },
-    ]),
+    assets: async ({ projectKey, assetIds }, args, { token }) => {
+      const assetMetadataRequests = assetIds
```

Comment:
Does this work on stacks where assetIds is not defined?  See for example how CentralAssetMetadata projects is handled, which doesn't require a Promise.all.

Reply:
Good point, I hadn't thought about that. I will update to make it handle assetIds not being defined. I hadn't seen that you could return an array of promises too. I will update to do that as it would be better to let Apollo handle that however it sees fit.